### ISSUE TITLE
Allow to sidestep plz-out

### ISSUE THREAD

**Issue by burdiyan** [thought-machine/please#1368]
This request comes from the frustration to implement build systems like Bazel and Please for rather complex projects, which are still not ready to immerse fully into the Bazel or Please semantics, like a forced separate output directory for everything.

I'm a Bazel user for about 2 years, and build-systems enthusiast in general, so I understand the benefits of Please and Bazel. But still, for 2 or 3 corner-cases I would really like to be able to sidestep the `plz-out`, at least using relative paths. 

For example, I'd like to check-in generated code from Protobuf files into the repo, so that IDEs and other tools can continue working without knowing about Please layout. But I still want the same behavior as with normal outputs (e.g. if output disappears, rerun the target, using outputs as srcs in other rules, etc).

Currently Please fails when `genrule` outs is declaring relative path with `trying to output file ../../out.txt, but that directory belongs to another package ()`.

**Comment by burdiyan** [comment]
Another, and probably better example for this is handling third party dependencies. Imagine a rule like this in the root `BUILD` file:

**Comment by burdiyan** [comment]
Actually, this probably can be generalized into "Allow to sidestep build isolation", and a better example could be a typical JS application using `node_modules`. Basically it's about letting one do whatever they Please (😀) if they know what they are doing :)

Fine-grained dependencies and separate execroots with exactly specified inputs are cool and bring lots of benefits, but sometimes cumbersome to work with, especially for newcomers. They are often slow for huge directory trees like `node_modules`.

I find that what people often want is "just": rebuild this target if these inputs change, or these outputs disappear + specify dependencies between targets. No matter where inputs and outputs live (separate out dir or in the source tree).

So, imagine a Git repo with multiple Yarn workspaces. Each directory is a separate package and they all depend on the top-level `node_modules`. For a typical project that uses frameworks like React, Babel, etc. this `node_modules` folder can be HUGE. Having a `filegroup` for this `node_modules` takes a lot of type copying these into the `plz-out/tmp` execroot for each target.

Would it be possible to configure Please to not use isolated execroots and have access to all the source tree, without breaking anything too much, or introducing lots of obscure code paths? Maybe a global configuration, or per-rule one? I know this doesn't fit the whole purpose of Please very well, but I really believe this can be useful to many folks out there, who are not Google scale, but have less than trivial projects with interdependencies, and don't want to use their normal tooling and IDEs.

**Comment by Tatskaari** [comment]
Before I say anything I'd like to say this is very much not advised. It breaks build isolation and you may find that Please is unable to correctly identify when to build/rebuild targets. You might find that things word when building all targets but when targetting individual targets, certain node modules might not exist. With that being said:

You can use ($out_location //path/to/target) in a build rule to find out the location of it's outputs within plz-out. There's nothing stopping you accessing these files within the build directory of a rule. We used to npm install our dependencies and used this to access the `node_modules` folder. This was very unprincipled and ended up being a huge amount of trouble.

We've since migrated to use the [yarn rules](https://github.com/thought-machine/pleasings/blob/master/js/yarn.build_defs) from the pleasings repo. These rules should enable you to use the standard yarn tooling during development (webpack dev server, yarn install etc.), but generate a principled set of build targets for Please. 

In summary, I understand your pain. JS builds are rather difficult to manage in a build system like Please. Please doesn't stop you from accessing things outside the build directory but it's almost certainly a bad idea.

**Comment by peterebden** [comment]
We have (had? can't find it now) a related issue about making the locations of `plz-out/bin` and `plz-out/gen` configurable. Would that help at all here?

**Comment by burdiyan** [comment]
Thanks @Tatskaari! I get your point!

@peterebden Nope, this won't help for this scenario.

Closing this as being irrelevant to the goals of this project.
